packages/neuron-ui/src/components/HardwareSign/index.tsx

```diff
@@ -315,12 +315,16 @@ const HardwareSign = ({
 
   const dialogClass = `${styles.dialog} ${wallet.isHD ? styles.hd : ''}`
 
-  const dropdownList = [
-    {
-      text: t('offline-sign.sign-and-export'),
-      onClick: signAndExportFromGenerateTx,
-    },
-  ]
+  const dropdownList = useMemo(() => {
+    return [
+      {
+        text: t('offline-sign.sign-and-export'),
+        onClick: signAndExportFromGenerateTx,
+        disabled: status === disconnectStatus,
+        disabledMsg: disconnectStatus,
+      },
+    ]
+  }, [signAndExportFromGenerateTx, status])
```

Comment:
`disconnectStatus` is listed in deps at at https://github.com/nervosnetwork/neuron/blob/refactor/disabled-sign-and-export-button/packages/neuron-ui/src/components/HardwareSign/index.tsx#L236 but not listed in deps here.

Reply:
I thought `disconnectStatus` wouldn't change, but after your reminder, it will change if the language setting is changed.